src/test/java/cat/udl/eps/softarch/mypadel/steps/CreateMatchStepDefs.java

```diff
@@ -0,0 +1,73 @@
+package cat.udl.eps.softarch.mypadel.steps;
+
+import cat.udl.eps.softarch.mypadel.domain.CourtType;
+import cat.udl.eps.softarch.mypadel.domain.Level;
+import cat.udl.eps.softarch.mypadel.domain.PublicMatch;
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import static cat.udl.eps.softarch.mypadel.steps.AuthenticationStepDefs.authenticate;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+public class CreateMatchStepDefs {
+
+    @Autowired
+    private StepDefs stepDefs;
+
+    private ZonedDateTime startDate = ZonedDateTime.of(2017, 10, 1, 0, 0, 0,
+                                                        0, ZoneId.of("+00:00"));
+
+    private Duration duration = Duration.ofMinutes(30);
+
+    private ZonedDateTime cancelationDeadline = ZonedDateTime.of(2017, 9, 30, 0, 0,
+                                                                 0, 0, ZoneId.of("+00:00"));
+
+    @When("^I create a new public match$")
```

Comment:
Accepting the corresponding PR but pending to include start and cancelation dates in the Cucumber text so they are step parameters instead of fixing the dates in the source code

Reply:
Got it!